flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/sinks/TableSink.scala

```diff
@@ -41,46 +38,13 @@ trait TableSink[T] {
     */
   def getOutputType: TypeInformation[T]
 
-  /** Return a deep copy of the [[TableSink]]. */
-  protected def copy: TableSink[T]
-
-  /**
-    * Return the field names of the [[org.apache.flink.api.table.Table]] to emit. */
-  protected final def getFieldNames: Array[String] = {
-    fieldNames match {
-      case Some(n) => n
-      case None => throw new IllegalStateException(
-        "TableSink must be configured to retrieve field names.")
-    }
-  }
-
-  /** Return the field types of the [[org.apache.flink.api.table.Table]] to emit. */
-  protected final def getFieldTypes: Array[TypeInformation[_]] = {
-    fieldTypes match {
-      case Some(t) => t
-      case None => throw new IllegalStateException(
-        "TableSink must be configured to retrieve field types.")
-    }
-  }
-
-  /**
-    * Return a copy of this [[TableSink]] configured with the field names and types of the
```

Comment:
Please keep the documentation for `configure()`. It is crucial that `configure()` returns a copy and not the same object.